### PR TITLE
Fan Control Spec Editorial Alignment

### LINKED ISSUE #27358: [FanControl] Align with spec editorial in Fan Control cluster
### Description

Following [this PR](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7210/) there are some small editorial items that need addressed in SDK. 

* The two bitmaps are now data types; `RockBitmap` & `WindBitmap`
* The anonymous enum types are now data types; `FanModeEnum` & `FanModeSequenceEnum`

### PR DESCRIPTION
**Pull request by mhazley**
### Description

@ReneJosefsen made some editorial changes to the spec which are addressed in this Pull Request. 

@bzbarsky-apple I think this will trigger some darwin changes which I'm not sure about, please let me know if you'd like me to make them as part of this PR.

### Links

Fixes #27358

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/fan-control-server/fan-control-server.cpp [hunk 1]
```diff
@@ -228,10 +229,11 @@ MatterFanControlClusterServerPreAttributeChangedCallback(const ConcreteAttribute
     case WindSupport::Id: {
         if (SupportsWind(attributePath.mEndpointId))
         {
-            uint8_t windSupport;
-            EmberAfStatus status = WindSupport::Get(attributePath.mEndpointId, &windSupport);
+            chip::BitMask<WindBitmap> windSupport;
+            EmberAfStatus status   = WindSupport::Get(attributePath.mEndpointId, &windSupport);
+            uint8_t rawWindSupport = windSupport.Raw();
```
Comment (bzbarsky-apple) [inline src/app/clusters/fan-control-server/fan-control-server.cpp:234]: See comments above for RockSupport.

That said (pre-existing, please file an issue) this code makes no sense.  It's constraining what you can set the "WindSupport" attribute to based on the _current_ value of that attribute.  The case should be `WindSetting::Id`, no?

File: src/app/clusters/fan-control-server/fan-control-server.cpp [hunk 2]
```diff
@@ -318,7 +320,7 @@ void MatterFanControlClusterServerAttributeChangedCallback(const app::ConcreteAt
         break;
     }
     case PercentSetting::Id: {
-        DataModel::Nullable<uint8_t> percentSetting;
+        DataModel::Nullable<chip::Percent> percentSetting;
```
Comment (bzbarsky-apple) [inline src/app/clusters/fan-control-server/fan-control-server.cpp:323]:
```suggestion
        DataModel::Nullable<Percent> percentSetting;
```
Going to stop commenting on this; just fix them all, please.

File: src/app/zap-templates/zcl/data-model/chip/fan-control-cluster.xml [hunk 3]
```diff
@@ -48,20 +48,14 @@ limitations under the License.
...
-    <cluster code="0x0202" />
-    <field name="Sleep Wind" mask="0x01" />
-    <field name="Natural Wind" mask="0x02" />
-  </bitmap>
-
-  <bitmap name="WindSettingMask" type="BITMAP8">
+  <bitmap name="WindBitmap" type="BITMAP8">
     <cluster code="0x0202" />
     <field name="Sleep Wind" mask="0x01" />
     <field name="Natural Wind" mask="0x02" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/fan-control-cluster.xml:61]: DirectionEnum needs to become StepDirectionEnum, no?  I know it was not part of the editorial PR but some other PR (which didn't file an SDK issue to track), but still.

**Comment by bzbarsky-apple** [comment]
Note: the test failures are because the type names in YAML need to be adjusted.

File: examples/all-clusters-app/all-clusters-common/src/fan-stub.cpp [hunk 4]
```diff
@@ -57,9 +57,9 @@ class FanAttrAccess : public AttributeAccessInterface
 CHIP_ERROR FanAttrAccess::ReadPercentCurrent(EndpointId endpoint, AttributeValueEncoder & aEncoder)
 {
     // Return PercentSetting attribute value for now
-    DataModel::Nullable<uint8_t> percentSetting;
+    DataModel::Nullable<chip::Percent> percentSetting;
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/all-clusters-common/src/fan-stub.cpp:60]:
```suggestion
    DataModel::Nullable<Percent> percentSetting;
```
and throughout....

File: src/app/clusters/fan-control-server/fan-control-server.cpp [hunk 5]
```diff
@@ -207,10 +207,11 @@ MatterFanControlClusterServerPreAttributeChangedCallback(const ConcreteAttribute
     case RockSetting::Id: {
         if (SupportsRocking(attributePath.mEndpointId))
         {
-            uint8_t rockSupport;
+            chip::BitMask<RockBitmap> rockSupport;
```
Comment (bzbarsky-apple) [inline src/app/clusters/fan-control-server/fan-control-server.cpp:210]:
```suggestion
            BitMask<RockBitmap> rockSupport;
```

File: src/app/clusters/fan-control-server/fan-control-server.cpp [hunk 6]
```diff
@@ -225,13 +226,14 @@ MatterFanControlClusterServerPreAttributeChangedCallback(const ConcreteAttribute
         }
         break;
     }
-    case WindSupport::Id: {
+    case WindSetting::Id: {
         if (SupportsWind(attributePath.mEndpointId))
         {
-            uint8_t windSupport;
+            chip::BitMask<WindBitmap> windSupport;
```
Comment (bzbarsky-apple) [inline src/app/clusters/fan-control-server/fan-control-server.cpp:232]:
```suggestion
            BitMask<WindBitmap> windSupport;
```

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**